### ISSUE TITLE
is it possible to use slackin now ? (legacy tokens are not supported by Slack anymore)

### ISSUE THREAD

**Issue by StefanoSalsano** [rauchg/slackin#416]
Slack no longer support the use of legacy tokens and replaces them with Slack apps.
This was already mentioned in issue #412 :

> It looks like you can still create legacy tokens until May 5th here: https://api.slack.com/legacy/custom-integrations/legacy-tokens

> After legacy tokens are deprecated, it seems only Enterprise Slack teams will be able to use auto-invite flows/apps, because the admin.users.invite api method is restricted: https://api.slack.com/methods/admin.users.invite

is there a way to use slackin now ?

**Comment by dschinkel** [comment]
Looks like now you can just generate your own url to share 

![Screen Shot 2020-05-12 at 1 39 09 AM](https://user-images.githubusercontent.com/338798/81647346-7c991f80-93f2-11ea-8fe4-51debd72dec5.jpg)

So slackin is dead because going to that generated url gives you essentially what slackin gave you
https://join.slack.com/t/wedotdd/shared_invite/zt-e6fv7idi-xJojTQxM6rmjUvMGPYbXmg
![Screen Shot 2020-05-12 at 1 49 52 AM](https://user-images.githubusercontent.com/338798/81647730-1fea3480-93f3-11ea-90c8-6bf16ae10d98.jpg)

Now all you need to do is direct users to that instead.

**Comment by jordanvanbergen** [comment]
> Looks like now you can just generate your own url to share
> 
> ![Screen Shot 2020-05-12 at 1 39 09 AM](https://user-images.githubusercontent.com/338798/81647346-7c991f80-93f2-11ea-8fe4-51debd72dec5.jpg)
> 
> So slackin is dead because going to that generated url gives you essentially what slackin gave you
> https://join.slack.com/t/wedotdd/shared_invite/zt-e6fv7idi-xJojTQxM6rmjUvMGPYbXmg
> ![Screen Shot 2020-05-12 at 1 49 52 AM](https://user-images.githubusercontent.com/338798/81647730-1fea3480-93f3-11ea-90c8-6bf16ae10d98.jpg)
> 
> Now all you need to do is direct users to that instead.

Invite links will expire due to security reasons. So you have I think 30 days and you need to recreate the invite link again.

**Comment by dschinkel** [comment]
where are you seeing that? Just wondering how you know that since the above does not mention that.

**Comment by jordanvanbergen** [comment]
> where are you seeing that? Just wondering how you know that since the above does not mention that.

![image](https://user-images.githubusercontent.com/11584671/84113080-356e7080-aa2a-11ea-9f91-4ee3ebbea984.png)

**Comment by dschinkel** [comment]
where is this even from?

**Comment by jordanvanbergen** [comment]
> where is this even from? This tells me nothing. What I showed you was the actual slack workspace that said it's not going to expire!

This has changed:

![image](https://user-images.githubusercontent.com/11584671/84113361-bc234d80-aa2a-11ea-8355-9ae3bdbd6a31.png)

And when you click share invite link:

![image](https://user-images.githubusercontent.com/11584671/84113419-d2c9a480-aa2a-11ea-814e-acde5d214f24.png)

So share invite links expire < 30 days.

**Comment by dschinkel** [comment]
Yea something's odd.  One of my workspaces shows it's not going to expire but a new workspace I just created, the shared link in that one does say it expires.  Now I'm really confused.

**Comment by dschinkel** [comment]
I'm attending the Slack call tomorrow where they talk about the roadmap.  If I can I'll ask about it.

**Comment by jordanvanbergen** [comment]
> I'm attending the Slack call tomorrow where they talk about the roadmap. If I can I'll ask about it.

It's possible ! I needed to deactive the link. Create a new one with a CUSTOM expire date and you can select never expire. So it's working with never expire.

**Comment by dschinkel** [comment]
ah ok that must be what I did different.  Interesting.

**Comment by Lewiscowles1986** [comment]
They should still provision for this to work via oauth bot scopes

**Comment by jordanvanbergen** [comment]
> They should still provision for this to work via oauth bot scopes

You say should. We can't make a legacy token any longer. So if someone knows how to setup a connection to the api that would be nice. Until then only not expiring invite links from within Slack itself are possilbe.

**Comment by Lewiscowles1986** [comment]
Yes that is the option I've gone for, archiving the fork of this repo

**Comment by Lewiscowles1986** [comment]
Long term will probably look for a chat software a bit more suited to needs

**Comment by dschinkel** [comment]
TBH, this is much easier (no offense to the creators) just using slack.  Slackin required a server for what I can just now do via native slack.

**Comment by dschinkel** [comment]
I don't know what Slack is doing.  Now it expires again in their latest build:

![Screen Shot 2020-08-23 at 7 50 10 PM](https://user-images.githubusercontent.com/338798/90993333-1dfb4a00-e57a-11ea-90f6-4e49eb791120.jpg)
![Screen Shot 2020-08-23 at 7 50 18 PM](https://user-images.githubusercontent.com/338798/90993335-205da400-e57a-11ea-97c8-afa491762199.jpg)

**Comment by sruti** [comment]
I was able to create a link that never expires today, not sure why the option exists for some and not others...

**Comment by dschinkel** [comment]
did you upgrade to the latest version of Slack? **4.8.0** took the option away

**Comment by dschinkel** [comment]
I opened basically a complaint on it here https://github.com/integrations/slack/issues/1063

**Comment by sruti** [comment]
> did you upgrade to the latest version of Slack? **4.8.0** took the option away

I used the webapp and not the desktop app, maybe that's the difference. I'm guessing I'm using the latest version.

**Comment by dschinkel** [comment]
Most people use the desktop client...

**Comment by slifty** [comment]
It appears that paid organizations can create invitations that never expire; free orgs have a 30 day expiration.

**Comment by softmarshmallow** [comment]
Is invitation link generation as api not requires enterprise version?, if so we can alt this project to generate it with cron job, redirect them on the existing ui.
